arbor/include/arbor/math.hpp

```diff
@@ -32,6 +32,19 @@ T constexpr area_circle(T r) {
     return pi<T> * square(r);
 }
 
+template <typename T, typename U, typename = std::enable_if_t<std::is_unsigned_v<U>>>
```

Comment:
Suggestion here:
```cxx
/// Exponentiation for integral exponents. Uses exponentiation by squaring
template <typename T, typename I, typename = std::enable_if_t<std::is_integral_v<I>>>
T constexpr pow(T base, I exp) {
    // 0^0 and 0^-n are undefined
    if (base == 0 && exp <= 0) return NAN;
    // trivial cases
    if (base == 0 || base == 1) return base;
    // negative exp
    if (exp < 0) {
       exp = -exp;
       base = 1/base;
    }
    // log2(exp)
    T acc = 1;
    for (exp > 0; exp >> 1) {
         if (exp & 1) {
              // NOTE: Could get cute here and eliminate the branch.
              acc *= base;
              exp -= 1;
         } 
         base *= base;
    }
    return acc;
}
```
1. it's iterative instead of recursive
2. allows more exponents
3. Simpler test for odd numbers

Reply:
I like this version in general. But the problem is that T might be an integer type, in which case returning NAN would not work. So I'd suggest keeping the unsigned restriction for an even simpler version:
```c++
template <typename T, typename U, typename = std::enable_if_t<std::is_unsigned_v<U>>>
T constexpr pow(T base, U exp) {
    // log2(exp)
    T acc = 1;
    for (;exp > 0; exp >>= 1) {
        if (exp & 1) acc *= base;
        base *= base;
    }
    return acc;
}
```